### DISCUSSION TITLE
(No Replacer tab) Error Loading Scripts

### DISCUSSION THREAD

**Discussion by tuanakotta** [light-and-ray/sd-webui-replacer#22] [Q&A] [answered]
I am using openvinotoolkit sd-webui repo instead of main a1111 sd-webui and got this error after installing sd-webui-replacer and restarting my SD. My SD worked well with other extension such as ADetailer, Controlnet and so on.
    
```
    *** Error loading script: replacer_main_ui.py
    Traceback (most recent call last):
      File "C:\stable-diffusion-webui\modules\scripts.py", line 382, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "C:\stable-diffusion-webui\modules\script_loading.py", line 10, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "C:\stable-diffusion-webui\extensions\sd-webui-replacer\scripts\replacer_main_ui.py", line 2, in <module>
        from modules import scripts, shared, sd_samplers, ui_toprow, ui, script_callbacks
    ImportError: cannot import name 'ui_toprow' from 'modules' (unknown location)
```

Do you know how to fix it?

**Comment by light-and-ray** [comment, marked as answer]
Known issues:

> ImportError: cannot import name 'ui_toprow' from 'modules' (unknown location)

You need to update automatic1111 webui. Launch `git pull` command inside webui root

> ModuleNotFoundError: No module named 'scripts.sam'

You need to install https://github.com/continue-revolution/sd-webui-segment-anything
Do not confuse with Inpaint Anything!

> sam = sam_model_registrymodel_type
> KeyError: ''

Segment Anything extension currently doesn't support FastSam and Matting-Anything. Ask about this here: https://github.com/continue-revolution/sd-webui-segment-anything/issues/135

https://github.com/light-and-ray/sd-webui-replacer/issues/16
